crates/activate/src/lib.rs

```diff
@@ -253,15 +268,30 @@ async fn apply_changes(
     Ok(())
 }
 
+/// Describes the desired future state of a task.
+/// Virtual tasks get logs and stats journals,
+/// but are otherwise purely descriptive and
+/// do not get shards and recovery log journals
+/// created for them like real tasks do.
+#[derive(Clone, Copy, Debug)]
+enum TaskTemplate<'a> {
```

Comment:
I don't _love_ the naming of real vs virtual tasks here, though I could live with it. I'll toss out a suggestion of `UpsertShard` and `UpsertOpsJournals` if that sounds any better to you.

Reply:
I think I actually prefer the Real / Virtual distinction, because UpsertShard also implies creation of ops journals, which isn't clear with UpsertopsJournals.

`UpsertShard` and `UpsertPlaceholder` is the other place my mind goes but I don't think that's better.